bool has_a_fault
